### ISSUE TITLE
Sphinx crashes on PySide2 and macOS 10.13

### ISSUE THREAD

**Issue by ctismer** [sphinx-doc/sphinx#9074]
**Describe the bug**
When running a PySide2 compilation with Sphinx doc generation, we get
an encoding error. The error does not happen with a recent macOS.

**To Reproduce**

Do a setup according to https://doc.qt.io/qtforpython/gettingstarted.html
Then install sphinx and run a compile.
Simulate the effect of old macOS without unicode setting.
```
$ pip3 -m pip install sphinx
$ export LC_ALL="C"
$ python3 setup.py build --parallel=8 --build-tests
```

**Expected behavior**
```
Generating Shiboken documentation
In directory /Users/tismer/src/QtC/tqtc-pyside-setup/pyside3a_build/py3.6-qt5.15.2-64bit-release/shiboken2:
    Running command:  /usr/bin/make doc
[100%] Generating HTMLs...
Running Sphinx v3.5.3

Encoding error:
'ascii' codec can't decode byte 0xc2 in position 69: ordinal not in range(128)
The full traceback has been saved in /var/folders/v_/0y4_1t552512kjkcfws0jf6c0000gn/T/sphinx-err-bw3djppf.log, if you want to report the issue to the developers.
make[3]: *** [doc/CMakeFiles/doc] Error 2
make[2]: *** [doc/CMakeFiles/doc.dir/all] Error 2
make[1]: *** [doc/CMakeFiles/doc.dir/rule] Error 2
make: *** [doc] Error 2
error: Error generating documentation for shiboken2
Traceback (most recent call last):
  File "setup.py", line 296, in <module>
    setup_runner.run_setup()
  File "/Users/tismer/src/QtC/tqtc-pyside-setup/build_scripts/setup_runner.py", line 168, in run_setup
    raise RuntimeError(msg)
RuntimeError: 
setup.py invocation failed with exit code: 1.
```
**Your project**
PySide2 / PySide6

**Environment info**
- OS: macOS 10.13, or newer with LC_ALL="C"
- Python version: 3.6
- Sphinx version: 3.5.3

**Additional context**

Remark:
This does not happen with Python 2 and the according old Sphinx version.
Also, the same setting of LC_ALL does not produce an error on Linux.

The obvious assumption that macOS always has unicode is wrong.
It is as vulnerable for encoding errors as any other Unix platform.

The problem is visible on a commercial branch, only, because there are copyright signs
in the license headers. This can be easily tested with a public repo by inserting a non-ascii char.

**Comment by tk0miya** [comment]
Could you share the full traceback file to us, please?
The full traceback has been saved in /var/folders/v_/0y4_1t552512kjkcfws0jf6c0000gn/T/sphinx-err-bw3djppf.log, if you want to report the issue to the developers.

Additionally, please check the version of docutils. At present, we have troubles with the docutils-0.17. If you're using it, please downgrade it to v0.16 for a while.

**Comment by ctismer** [comment]
The whole problem only happens since we have such an old CI implementation and cannot update ATM.

I just was hit by the fact that the whole problem does only occur in a non-public branch,
and it seems to happen with Python 3.6, only.

I will do a more exact analysis before coming back on you. Possibly it is not even a Sphinx error.
Thanks!

**Comment by ctismer** [comment]
[sphinx-err-4td5savd.log](https://github.com/sphinx-doc/sphinx/files/6289580/sphinx-err-4td5savd.log)
This was with Python 3.6.13. With Python 3.7.0, the error is gone.

The hint with docutils v0.16 helped, great of course!
The question remains: Why does this happen with Python 3.6, only?

**Comment by tk0miya** [comment]
Oops... this is a bug of docutils-0.17. The root cause of the error is the default character encodings settings of Python. Before 3.7, ASCII is used for it. On the other hand, since 3.7, utf-8 is used for it by default in Linux (for more details, see PEP 538). As a result, docutils fails to read an internal file on python3.6 or older.

So please pin the docutils to 0.16 for a while (0.17.1 will be released soon), or upgrade your python to 3.7+.
Thanks,

**Comment by ctismer** [comment]
Yes, I just found the Python 3.7 change, too.
Why is that not an issue on Python 2.7? Ok, that is a much older Sphinx version.

I think we will set a default encoding of utf-8 for CI. That will always be ok,
regardless what docutils thinks.
We are hopefully removing Python 3.6 when 3.10 is out.
Cheers, and thanks for your help!
